cunumeric/deferred.py

```diff
@@ -781,10 +781,16 @@ def _broadcast(self, shape: NdShape) -> Any:
 
         return result
 
-    def _convert_future_to_regionfield(self) -> DeferredArray:
+    def _convert_future_to_regionfield(
+        self, future: bool = False
+    ) -> DeferredArray:
+        if future:
+            shape: NdShape = (1,)
+        else:
+            shape = self.shape
```

Comment:
Why do you need to do this? It looks like you're overriding the shape of the target array if it's backed by a future. Did you see a case where a future-backed array had the wrong shape?

Reply:
Yes, for the case when scalar is converted to the ndarray it will get `()` shape. But we actually want it to be `(1,)` shape

Reply:
That makes sense, however I would it to be more explicit. Can we just check in `_convert_future_to_regionfield` whether `self.shape == ()`, and in that case do `shape = (1,)`? That would make the purpose of this obvious. In that case I think you don't need the `future: bool` argument.

Reply:
yes, this is rather a special case for the `wrap`. I will look how to change the logic in wrap so that we don't have to add this change to  `_convert_future_to_regionfield`

Reply:
@manolis, actually I don't know how to do this without converting the shape of the futures for put routine. We need to do this not for the wrap task, but for the copy operation after the task. I can convert shape of the indirect field only in the task, but not the shape of indices or values there.
how about renaming this boolean variable to "change_shape" instead of the "future" in the `_convert_future_to_regionfield`?

Reply:
That's fine, I think it would be cleanest to check in `_convert_future_to_regionfield` something like:
```
if change_shape and shape == ():
  shape = (1,)
```
and pass `change_shape=True` whenever this is relevant. Then it's obvious that we're only changing `()` to `(1,)`, and not other shapes.

Reply:
Sounds good! thank you